SonarRunner.Shim/RoslynV1SarifFixer.cs

```diff
@@ -124,5 +129,38 @@ public static bool FixRoslynV1Sarif(string unfixedSarif, out string output)
                 }
             }            
         }
+
+        /// <summary>
+        /// Attempts to load and fix a SARIF file emitted by Roslyn 1.0 (VS 2015 RTM).
+       ///  Returns true if the file at the given path is valid a JSON file suitable for uploading to the server.
+        /// </summary>
+        public bool FixRoslynV1SarifFile(string sarifFilePath, ILogger logger)
+        {
+            // If the file cannot be found it is impossible to fix it into valid JSON
+            if (!File.Exists(sarifFilePath))
+            {
+                logger.LogInfo(Resources.MSG_SarifFileNotFound, sarifFilePath);
+                return false;
+            }
+
+            string inputSarifFileString = File.ReadAllText(sarifFilePath);
+            string outputSarifFileString;
+
+
+            bool fixSuccess = FixRoslynV1SarifString(inputSarifFileString, out outputSarifFileString);
+
+            if (fixSuccess)
+            {
+                // Write out the fixed JSON
+                File.WriteAllText(sarifFilePath, outputSarifFileString);
```

Comment:
We discussed whether it would be better to save the fixed SARIF to a different file to make troubleshooting easier (and as @bgavrilMS pointed out it would also mean that we couldn't impact any other tools that consumed the invalid file). Was it a deliberate decision to overwrite the existing file?